calculator.rb

```diff
@@ -0,0 +1,98 @@
+#Setting Up Operators
+operators = ["1", "add", "+", "2", "subtract", "-", "3", "multiply", "*", "4", "divide", "/", "5", "exponent", "^", "6", "modulo", "remainder", "%"]
+numbers = []
+
+#Validation Methods
+def data_type (num_one)
+    if num_one.include?(".") == true
+        num_one = num_one.to_f
+    else
+        num_one = num_one.to_i
+    end
+    return num_one
+end
+
+def number?(obj)
+    obj = obj.to_s unless obj.is_a? String
+    /\A[+-]?\d+(\.[\d]+)?\z/.match obj
+    #Lines 15-17 Source: https://medium.com/launch-school/number-validation-with-regex-ruby-393954e46797
+end
+
+#Setting Up Calculation Methods
+def addition (num_one, num_two)
+    print "#{num_one} + #{num_two} = "
+    return num_one + num_two
+end
+
+def subtraction (num_one, num_two)
+    print "#{num_one} - #{num_two} = "
+    return num_one - num_two
+end
+
+def multiplication (num_one, num_two)
+    print "#{num_one} * #{num_two} = "
+    return (num_one * num_two)
+end
+
+def division (num_one, num_two)
+    if num_one == 0 || num_two == 0
+        puts "Invalid Result - Cannot Divide by 0"
+    else
+        return (num_one / num_two)
+        print "#{num_one} / #{num_two} = "
+    end
+end
+
+def exponent (num_one, num_two)
+    print "#{num_one} ^ #{num_two} = "
+    return num_one ** num_two
+end
+
+def remainder (num_one, num_two)
+    print "#{num_one} % #{num_two} = "
+    return (num_one % num_two).to_i
+end
+
+#User Main Menu
+puts "Calculator Program\n\nWhich Operator would you like to Use?"
+puts "1. Add(+)\n2. Subtract(-)\n3. Multiply(*)\n4. Divide(/)\n5. Exponent(^)\n6. Modulo or Remainder(%)"
+
+print "\nPlease choose one operator(name or symbol): "
+operator = gets.chomp.downcase.to_s
+
+#User Operator Input Check
+until operators.include?(operator)
```

Comment:
You put other validation functionality into their own methods. Might you use a method here to encapsulate this user validation?